### PR TITLE
ENH: add a maxiter argument to NNLS

### LINKED ISSUE #4161: Missing ITMAX optional argument in scipy.optimize.nnls
The Fortran code of NNLS includes a parameter which specifies the maximum number of iterations before an error is raised. This parameter is hard-coded by default in the Fortan code as ITMAX=3*N, but the user is encouraged to modify it an recompile when needed. In fact this default is not sufficient in many real-world situations. I myself often found the need to increase ITMAX when using the Fortran code.

The newer and more general version of NNLS, written by the same authors and called BVLS, includes the same parameters. The authors of the Fortran code write in the documentation [here](http://www.netlib.org/lawson-hanson/all):

```
!   ITMAX  [integer]
!       Set to 3*N.  Maximum number of iterations permitted.
!       This is usually larger than required.  Library software will 
!       likely make this an optional argument.
```

In the Scipy wrapper of NNLS an error is raised when the maximum number of iterations has been exceeded, but when that happens one cannot increase ITMAX, as the parameter has not been exposed as one of nnls's optional parameters, as recommended above by the NNLS authors.

ITMAX should be included as an optional argument for nnls(), with default value itmax=3*n, if None is provided.

### PR DESCRIPTION
**Pull request by ev-br**
As an alternative to https://github.com/scipy/scipy/pull/8323, this adds an optional `maxiter` argument to `nnls`.

closes https://github.com/scipy/scipy/issues/4161

### CODE REVIEW

File: scipy/optimize/nnls/nnls.f [hunk 1]
```diff
@@ -44,13 +44,16 @@
...
 C             2     THE DIMENSIONS OF THE PROBLEM ARE BAD.  
 C                   EITHER M .LE. 0 OR N .LE. 0.
-C             3    ITERATION COUNT EXCEEDED.  MORE THAN 3*N ITERATIONS. 
+C             3    ITERATION COUNT EXCEEDED.  MORE THAN MAXITER ITERATIONS.
+C     MAXITER THE MAXIMUM ALLOWED NUMBER OF ITERATIONS.
+C             IF NEGATIVE, THE LIMIT IS TAKEN AS THE DEFAULT, 3*N.              
 C   
 C     ------------------------------------------------------------------
-      SUBROUTINE NNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,MODE) 
+      SUBROUTINE NNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,MODE, MAXITER)
```
Comment (tylerjereddy) [inline scipy/optimize/nnls/nnls.f]: I'm not sure what the Fortran standard is, and this is a minor comment, but just noticing that here and elsewhere there's no space after comma between subroutine parameters except for the new parameter added.

**Review by tylerjereddy — COMMENTED** [review]
This looks like a solid improvement & I like the creative unit test to prevent regression of the enhanced functionality. 

Let's see what Travis says & make sure the circle CI compile of docstring adjustments look ok.

**Review by tylerjereddy — APPROVED** [review]
My minor comments were addressed, the CI tests all pass, and the new docstring looks ok in the build artifact, so in it goes.

**Comment by andyfaff** [comment]
On macOS I'm getting a little test cruft, not sure if it's because of this?

```
============================================= 701 passed, 2 xfailed, 2 xpassed in 17.31 seconds =============================================

   Normal return from subroutine COBYLA

   NFVALS =   50   F = 2.485185E+01    MAXCV = 1.999965E-10
   X = 4.955358E+00   6.666553E-01

 NNLS quitting on iteration count.
```

**Pull request merged**